### DISCUSSION TITLE
How to handle wifi-access (and more) "in the field"?

### DISCUSSION THREAD

**Discussion by joknjokn** [mcguirepr89/BirdNET-Pi#119] [Q&A]
Hi,

I want to try BirdNET-pi in sessions in the field, running from a powerbank.

I'm now trying to figure out how to handle wifi/access to the RPi. Any experiences with this?

I tried "sudo raspi-config" to connect the RPi to my smartphone hotspot hoping I could open birdnetpi.local from that, but it doesn't work. However, it does have a connection. I used an app to find the IP of the RPi on my phones' network, and opening that ip with :8080 opened the green BirdNET-Pi log screen. :80 just showed a white page. So, not really a solution yet. I installed BirdNET-Pi by first installing RaspiOS, then the "newinstaller".

I think the optimal solution would be:
1. The website (birdnetpi.local) should work when connected to a phone-hotspot as well (no clue if possible?)
2. If the RPi doesn't find a network, it creates its' own hotspot (how to handle password?) where birdnetpi.local works. I think I saw this mentioned as a feature in another thread, but unsure how to configure it, or if it's even available with my installation method?

Bonus info:
Just got my RPi 4 2gb mounted in an Argon Neo case without a fan. Running at 48-51 degrees. Very smooth! Hoping to run it with a good powerbank in the field.

**Comment by joknjokn** [comment]
Also: When turning off the hotspot of my phone, the RPi doesn't automatically connect to my home wifi afterwards. 
UPDATE: It actually did reconnect to my home wifi after I rebooted the RPi over ssh from my phone and then disabled the hotspot on my phone.

**Comment by joknjokn** [comment]
A few other ideas for use in the field (I guess it's a suggestion to add something to the documentation, if anyone has a solution... I will probably try at some point):
- Adding an indicator LED that shows if BirdNET-PI is running (and working). The RPi takes some time to boot, and things can go wrong, so a simple indicator would be nice. I could probably connect an LED myself, but unsure how to turn it on when BirdNET-Pi is running.
- Shutdown-button to safely shut the RPi down instead of unplugging a powerbank. I guess I can find a bunch of tutorials for this.

This would allow for field sessions without messing around with phone/hotspot/wifi/browser-stuff.

**Comment by nilspupils** [comment]
Hi @joknjokn, just a quick info that most of your issues will be resolved by using the pre-installed image, which offers a hotspot when no network is found. Please have a look here: https://github.com/mcguirepr89/BirdNET-Pi/discussions/11#discussioncomment-1751201
Btw. i like your idea with the LED as a simple status display. This could be used for the other functions as well.
About powering the Pi with a powerbank this will be difficult as the Pi is quite power hungry. Even with a large powerbank only a couple of hours will be possible at best.
Also, have a look here about turning off BirdNET.-Pi: https://github.com/mcguirepr89/BirdNET-Pi/discussions/11#discussioncomment-1791801
I could not get it to work yet (but that's probably just my system).

**Comment by mcguirepr89** [comment]
I am coming back to this with a nice long response, stay tuned!

**Comment by mcguirepr89** [comment]
Hi, @joknjokn and thank you for the great question. Thanks, too, to @nilspupils for pointing out where we've discussed this possibility before.

@nilspupils is correct that the exact feature you're hoping for was already included in the pre-installed image for version 0.10. Version 0.11 (currently known as the testing `forms` branch) will be including an interface to enable this auto-hotspot feature manually. Once I've really finished development on v0.11, I will create a new pre-installed image for it that will include the auto-hotspot feature.

The basics are that when you first boot up the pre-installed image, it will present its own hotspot to which you would connect in the field. From that connection, you would set the system time (since RPis don't have RTCs built in) and then give it the LAT and LON it needs to get started. From that hotspot, you have full access to the installation. Likewise, you can use that hotspot connection to configure a WiFi connection to a WiFi network. The goal is that the pre-installed image would work as both a quick way to get the installation up and running, but also a way for fully remote field work to function as needed without the need for a host network other than the one the BirdNET-Pi can provide.

You can take a look at the software that I use for this feature [here](https://github.com/gitbls/autoAP). I customize it a bit before making it part of the base image, but it works well and can be used as is.

>Adding an indicator LED that shows if BirdNET-PI is running (and working). The RPi takes some time to boot, and things can go wrong, so a simple indicator would be nice. I could probably connect an LED myself, but unsure how to turn it on when BirdNET-Pi is running.

This would be very easy to implement using the GPIOs, but it is no longer (reasonably) possible to control the Pi's built in LEDs. If you put something together that uses this (probably using the python RPi.GPIO library), let me know and I can work with you on getting it to indicate whatever you need.

My best regards,
Patrick

**Reply by davidmam** [reply]
I've been experimenting with this using a cellular modem on the RasPi and a hotspot on my phone.

1. If you have previously set up the pi on a wifi network, faking it [1] with your phone works and you can get to hostname.local via phone or PC. It will then rejoin the home network when brought back in range.
2. It eats power when running TF and a cellular modem. I am using a PiJuice and the stock 1800 mAh battery lasts about 40 mins at best. I do have a 30Ah powerbank which I use for overnight audio recording which should then give overnight but would need recharging. 
3. Yet to get it to update as i have an issue with the birdnet server not running properly. I might have to wipe and restart with the latest image. Currently in the shed on mains power.

[1] ie set the SSID and password to be the same. I found avahi was fine with updating that, having also had to play with nmap to track down the host.

**Reply by mcguirepr89** [reply]
Configuring your phone's hotspot to have the same SSID and Password as your home network is quite brilliant.

**Reply by joknjokn** [reply]
@davidmam Any idea why the main page birdnetpi.local doesn't work (in phone browser) when the Pi is connected to by phone's hotspot?

**Reply by mcguirepr89** [reply]
Just a note/update: I'm going to be working on integrating `autoAP` soon so that the BirdNET-Pi is the one issuing the hotspot, and it will indeed support mDNS.

**Comment by mcguirepr89** [comment]
## Changing this to a discussion

**Comment by davidmam** [comment]
No. It depends on your phone hotspot supporting avahi type naming. Mine does so it worked.

..d
--

[University of Dundee shield logo]<http://uod.ac.uk/sig-home>


Dr David Martin
Programme Lead Core Curriculum
School of Life Sciences, University of Dundee
+44 (0)1382 388704 | ***@***.******@***.***>
[University of Dundee Facebook]<http://uod.ac.uk/sig-fb>[University of Dundee Twitter]<http://uod.ac.uk/sig-tw>[University of Dundee LinkedIn]<http://uod.ac.uk/sig-li>[University of Dundee YouTube]<http://uod.ac.uk/sig-yt>[University of Dundee Instagram]<http://uod.ac.uk/sig-ig>[University of Dundee TikTok]<http://uod.ac.uk/sig-sc>
One of the UK's top 20 universities<http://uod.ac.uk/sig-strapline>
The Guardian University Guide 2021
[Wear a face covering, Keep your distance, Wash your hands, Do not organise or attend parties or large social gatherings, Follow the signs and guidance]<http://uod.ac.uk/sig-cvc>

**Comment by DD4WH** [comment]
My plan is to go into the field for a week and take a BirdNET-Pi with me without any Wifi access.
So I found this thread -thank you all for the hints!- and it worked!
Here is a step-by-step workflow (maybe you would like to put this into the WIKI, Patrick ? @mcguirepr89 )

* etch the usual 64-bit LiteOS on an SD card using Raspberry Pi Imager and using your home Wifi connection SSID and password
* connect to BirdNET-Pi on your home Wifi and use the one-line-installer to install BirdNET-Pi [the installed version is from May, 4th, 2022]

From now on, every step can/should be executed in-the-field (without access to your home Wifi router)
* configure mobile hotspot on your smartphone/tablet to use exactly the same SSID and password as your home Wifi router (brilliant idea by davidmam https://github.com/mcguirepr89/BirdNET-Pi/discussions/119#discussioncomment-2128872)
* switch on hotspot of your smartphone
* BirdNET-Pi should now connect automagically to your smartphone and should be visible as a device in the mobile hotspot settings menu
* for iOS users, it might already work at this point by using `birdnetpi.local` in the Safari browser. However, Android users have to perform the following additional steps:
* write down the local IP adress of your BirdNET-Pi that is displayed in the mobile hotspot settings menu
* ensure you have Juicy SSH or any other SSH client installed on your smartphone
* connect to BirdNET-Pi via your SSH client on your smartphone using the local IP and user name / password
* in the terminal, issue: `sudo nano birdnet.conf`
* change the line `BIRDNETPI_URL= http://MY.LOCAL.IP.ADRESS`
* save with CTRL+O, exit with CTRL+X
* in the terminal, issue `sudo update_caddyfile.sh`
* after a while, the BirdNET-Pi can be reached in your smartphones browser using the local IP adress
